metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaHook.java

```diff
@@ -89,4 +89,13 @@ public void rollbackDropTable(Table table)
    */
   public void commitDropTable(Table table, boolean deleteData)
     throws MetaException;
+
+  /**
+   * Called after successfully after INSERT [OVERWRITE] statement is executed.
+   * @param table table definition
+   * @param overwrite true if it is INSERT OVERWRITE
+   *
+   * @throws MetaException
+   */
+  public void commitInsert(Table table, boolean overwrite) throws MetaException;
```

Comment:
Also should be add the pre/rollback even if there is no actual call to those methods from the hive side ? that would be misleading thought ?